src/V3Ast.h

```diff
@@ -1316,14 +1317,11 @@ class VNodeMatcher VL_NOT_FINAL {
 // The if() makes it faster, even though prefetch won't fault on null pointers
 #define ASTNODE_PREFETCH(nodep) \
     do { \
-        if (nodep) { \
-            VL_PREFETCH_RD(&((nodep)->m_nextp)); \
-            VL_PREFETCH_RD(&((nodep)->m_type)); \
-        } \
+        if (nodep) { nodep->prefetch(); } \
```

Comment:
I don't want a function call here because when using -O0 it makes a huge performance hit.  Perhaps we have #define ASTNODE_PREFETCHN (for NULL) which calles #define ASTNODE_PREFETCH now without the if()?

Reply:
I sympathise, but m_nextp and m_type are private fields, so the macro would have to hack around that which is a bit messy. I could always just change it to prefetch `reinterpret_cast<char*>(nodep) + 8` and the same + 64 bytes (or whatever cache line size we want to bake in). Alternatively in GNU C I can add `always_inline` and `optimize("-O3")` attributes to the method so it is never a call. Any preference?

Reply:
FWIW probably some other AstNode accessors like op*p/nextp/brokeExists would also well with always_inline..